src/CMakeLists.txt

```diff
@@ -1,60 +1,85 @@
 
-include(QbtTargetSources)
-
-find_package(Boost ${requiredBoostVersion} REQUIRED)
-find_package(LibtorrentRasterbar ${requiredLibtorrentVersion} REQUIRED)
-find_package(OpenSSL ${requiredOpensslVersion} REQUIRED)
+# defines used in multiple targets, so they are added globally here
+add_compile_definitions(QBT_VERSION_MAJOR=${PROJECT_VERSION_MAJOR})
+add_compile_definitions(QBT_VERSION_MINOR=${PROJECT_VERSION_MINOR})
+add_compile_definitions(QBT_VERSION_BUGFIX=${PROJECT_VERSION_PATCH})
+add_compile_definitions(QBT_VERSION_BUILD=${PROJECT_VERSION_TWEAK})
+set(PROJECT_VERSION "${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}.${PROJECT_VERSION_TWEAK}${QBT_VER_STATUS}")
+add_compile_definitions(QBT_VERSION="v${PROJECT_VERSION}")
+add_compile_definitions(QBT_VERSION_2="${PROJECT_VERSION}")
+add_compile_definitions(QT_DEPRECATED_WARNINGS)
+add_compile_definitions(QT_NO_CAST_TO_ASCII)
+add_compile_definitions(QT_USE_QSTRINGBUILDER)
+add_compile_definitions(QT_STRICT_ITERATORS)
 
-if (Boost_VERSION VERSION_LESS 106000)
-    add_definitions(-DBOOST_NO_CXX11_RVALUE_REFERENCES)
+if (NOT COUNTRIES_RESOLUTION)
+    add_compile_definitions(DISABLE_COUNTRIES_RESOLUTION)
 endif()
-
-find_package(Qt5 ${requiredQtVersion} REQUIRED COMPONENTS Core Network Xml LinguistTools)
-if (GUI)
-    find_package(Qt5Widgets ${requiredQtVersion} REQUIRED)
-    find_package(Qt5DBus ${requiredQtVersion})
+if (NOT GUI)
+    add_compile_definitions(DISABLE_GUI)
 endif()
-
-set_package_properties(Qt5DBus PROPERTIES
-    DESCRIPTION "Qt5 module for inter-process communication over the D-Bus protocol"
-    PURPOSE "Enables communication with other system components (e.g. notification service) via D-Bus. "
-    TYPE RECOMMENDED
-)
-
-set(CMAKE_AUTOMOC True)
-list(APPEND CMAKE_AUTORCC_OPTIONS -compress 9 -threshold 5)
-if (APPLE)
-    # Workaround CMake bug (autogen does not pass required parameters to moc)
-    # Relevant issue: https://gitlab.kitware.com/cmake/cmake/issues/18041
-    list(APPEND CMAKE_AUTOMOC_MOC_OPTIONS -DQ_OS_MACOS -DQ_OS_DARWIN)
+if (NOT WEBUI)
+    add_compile_definitions(DISABLE_WEBUI)
 endif()
 
-include_directories(${CMAKE_CURRENT_SOURCE_DIR})
+add_compile_definitions("$<$<NOT:$<CONFIG:Debug>>:QT_NO_DEBUG_OUTPUT>")
 
-# defines
-add_definitions(-DQT_DEPRECATED_WARNINGS)
-add_definitions(-DQT_NO_CAST_TO_ASCII)
-add_definitions(-DQT_USE_QSTRINGBUILDER)
-add_definitions(-DQT_STRICT_ITERATORS)
+# include directories - ideally, would be done per target instead of global directory scope
+include_directories(${CMAKE_CURRENT_SOURCE_DIR})
 
-if (CMAKE_BUILD_TYPE MATCHES "Debug")
-    message(STATUS "Project is built in DEBUG mode.")
-else()
-    message(STATUS "Project is built in RELEASE mode.")
-    message(STATUS "Disabling debug output.")
-    add_definitions(-DQT_NO_DEBUG_OUTPUT)
+# Windows-specific stuff
+if (WIN32)
+    include(winconf)
 endif()
 
-configure_file(config.h.cmakein ${CMAKE_CURRENT_BINARY_DIR}/config.h)
```

Comment:
Isn't it needed anymore?

Reply:
@glassez 

Nope. This file was a hack used to map the cmake defines to qBittorrent defines.

As an example, it had stuff like:

```cpp
#ifndef QBT_USE_GUI
#define DISABLE_GUI
#define DISABLE_COUNTRIES_RESOLUTION
#endif
```

So if `QBT_USE_GUI` was not set to `ON` in cmake, `DISABLE_GUI` and `DISABLE_COUNTRIES_RESOLUTION` would be defined for the purposes of C++ files preprocessing/compilation.


Now we just do:

```cmake
if (NOT GUI)
    add_compile_definitions(DISABLE_GUI) # ideally will be target_compile_definitions() for separate targets in the future
```